relayer/pkg/starknet/utils.go

```diff
@@ -71,15 +71,22 @@ func DecodeFelts(felts []*big.Int) ([]byte, error) {
 
 	data := []byte{}
 	buf := make([]byte, chunkSize)
-	length := int(felts[0].Int64())
+	length := felts[0].Uint64()
 
 	for _, felt := range felts[1:] {
```

Comment:
@archseer looks like we have one extra felt that we can remove from the ocr config if we want to optimise.

When we encode a byte array as felts, first felt represents the length. But when we set the contract config we prepend the length one more time so it basically [looks like this](https://github.com/smartcontractkit/chainlink-starknet/blob/3df46ca75401919aee6f9b65675ec48792a614c1/relayer/pkg/chainlink/ocr2/config_digester.go#L85):  
```
...[version][length][length][off_chain_config_raw]...
```

Reply:
It's two different lengths. The first is a number of felts, used by cairo/starknet to specify an array. The second is a byte length of the data after decoding, that tells us if there's any padding on the last felt. So a 65 byte array is encoded as 

```
[version] 3 65 x y z
```

Where felts = 3, byte length = 65 and `z` will contain three bytes and the rest is padding.

Reply:
Got it, thx!